elasticdl/python/master/evaluation_service.py

```diff
@@ -220,14 +220,21 @@ def try_to_create_new_job(self):
                 return True
         return False
 
-    def add_evaluation_task_if_needed(self, master_locking):
+    def add_evaluation_task_if_needed(self, master_locking, model_version):
         """
         Add step-based evaluation task
         """
-        model_version = self._master_servicer.get_model_version()
-        if self._eval_steps and model_version % self._eval_steps == 0:
+        if not model_version:
+            model_version = self._master_servicer.get_model_version()
+        if (
+            self._eval_steps
+            and model_version % self._eval_steps == 0
```

Comment:
In async mode, the version in each PS may be not the same. For example, PS 0 has a version 98; PS 1 has a version 99. The evaluation step is set to 100. 

Then PS 1 reach 100, it reports to master, master will start an evaluation task. Then PS 0 reach 100 later, master will start the evaluation task again.

Reply:
Yes, thus I check that whether model version is bigger than the last evaluation version, please refer to 

https://github.com/sql-machine-learning/elasticdl/pull/1469/files/6238237d4165aa380497041f3646b641de0a9a7a#diff-882613b085e647dc904b962e6fd1e8c7R232

In my opinion, this condition can solve this problem.

Reply:
Or we could check the model version in master like this:

```python
model_version = 0
i = 1
if model_version ==  i * self._eval_steps:
    i += 1
    start_evaluation()
```

Reply:
@LiMinghao1994 Got it, thanks.